flood_adapt/config.py

```diff
@@ -1,78 +1,180 @@
 from platform import system
-from typing import Any, Dict, Union
+from typing import ClassVar
 
 import tomli
-from pydantic import Field, computed_field, model_validator
+from pydantic import Field, computed_field, field_validator, model_validator
 from pydantic_settings import BaseSettings, SettingsConfigDict
 
 
 class Settings(BaseSettings):
+    """
+    The configuration settings for the FloodAdapt database and integrator.
+
+    Precedence is as follows: user arguments > environment variables > defaults in this class.
+    When loading is done, the settings are validated and the environment variables are updated with the loaded values.
+
+    If any required settings are missing or invalid, a ValidationError is raised.
+
+    Usage
+    -----
+    from flood_adapt.config import Settings
+
+    # One of the following:
+        # 1) Load settings from environment variables, if no environment variables are set, use defaults defined in the class:
+        settings = Settings()
+
+        # 2) Load settings from a .toml file, overwriting any environment variables set:
+        settings = Settings.read(toml_path: Path)
+
+        # 3) Load settings from keyword arguments, overwriting any environment variables:
+        settings = Settings(database_root="path/to/database", database_name="database_name", system_folder="path/to/system_folder")
+
+    Attributes
+    ----------
+    database_name : str
+        The name of the database.
+    database_root : Path
+        The root directory of the database.
+    system_folder : Path
+        The root directory of the system folder.
+
+    Properties
+    ----------
+    database_path : Path
+        The full path to the database.
+    sfincs_path : Path
+        The path to the SFINCS binary.
+    fiat_path : Path
+        The path to the FIAT binary.
+
+    Raises
+    ------
+    ValidationError
+        If required settings are missing or invalid.
+    """
+
+    SYSTEM_SUFFIXES: ClassVar[dict[str, str]] = {
+        "Windows": ".exe",
+        "Linux": "",
+        "Darwin": "",
+    }
+
     model_config = SettingsConfigDict(
         env_ignore_empty=True, validate_default=True
     )  # empty env uses default
 
-    database_root: Path = Field(default=Path(__file__).parents[2] / "Database")
-    system_folder: Path
-    database_name: str = Field(default="default")
-    sfincs_path: Path
-    fiat_path: Path
+    database_root: Path = Field(
+        default=Path(__file__).parents[2] / "Database", env="DATABASE_ROOT"
+    )
+    database_name: str = Field(default=None, min_length=1, env="DATABASE_NAME")
+    system_folder: Path = Field(default=None, env="SYSTEM_FOLDER")
+
+    @computed_field
+    @property
+    def sfincs_path(self) -> Path:
+        return self.system_folder / "sfincs" / f"sfincs{Settings._system_extension()}"
+
+    @computed_field
+    @property
+    def fiat_path(self) -> Path:
+        return self.system_folder / "fiat" / f"fiat{Settings._system_extension()}"
```

Comment:
Now your `sfincs_path` and your `fiat_path` always have to be the same relative to your system folder. Why not make them fully configurable?

Reply:
Well, the idea of the system folder is to contain the model binaries. If we leave the model paths fully configurable, we might as well remove the system folder as a whole.

Reply:
Why not? :)

Reply:
Now that I think about it, I dont see any use case for changing only one the model_paths or even the system folder in general. 

The only reason it is configurable in the first place is that it is stored on the FloodAdapt svn, and thereby the default location was in the database root, but that was a problem for distribution, as that meant users could accidentally delete/change/be confused, so it was made configurable and moved to _internal. But now there is this random dir floating around, that is critical for FloodAdapt to run and we can configure it, but why do we even want to.

I think it shouldnt even be configurable in the first place and we should just hard code it (or leave it configurable, but change the default) to expect the system folder in the source code (flood_adapt/system) since it is actually a critical part of the backend and without it, nothing works. 

This will make everyone's life easier:
- We can include the system folder when we make a release. (either already in the right place, or as a .zip attached)
- New models can still be added by just adding them to the system folder and setting up the path.
- When we create releases for windows & linux, we can include the correct system folder corresponding to the os.
- When we create exe / installers for windows & linux we do the same.

Reply:
I would definitely package executables with the application.
If I have a working sphinx version on my workstation I can create a soft-link in this folder and make it work, so no worries there. With your current code, my initial problem of directly referencing a `sfincs.exe` is solved!